test/src/receiver/main.cpp

```diff
@@ -28,102 +28,148 @@ using TIME = NDTime;
+    using
+        info=cadmium::logger::logger<cadmium::logger::logger_info,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        debug=cadmium::logger::logger<cadmium::logger::logger_debug,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        state=cadmium::logger::logger<cadmium::logger::logger_state,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        log_messages=cadmium::logger::logger<cadmium::logger::logger_messages,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        routing=cadmium::logger::logger<cadmium::logger::logger_message_routing,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        global_time=cadmium::logger::logger<cadmium::logger::logger_global_time,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        local_time=cadmium::logger::logger<cadmium::logger::logger_local_time,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        log_all=cadmium::logger::multilogger <info,
+            debug, state, log_messages, routing, global_time, local_time>;
+
+    using logger_top=cadmium::logger::multilogger<log_messages, global_time>;
+
+
+    /*******************************************/
+
+
+
+    /********************************************/
+    /****** APPLICATION GENERATOR *******************/
+    /********************************************/
+    string input_data_control =
+        "../test/data/receiver/receiver_input_test.txt";
+    const char * i_input_data_control = input_data_control.c_str();
+
+    std::shared_ptr<cadmium::dynamic::modeling::model>
+        generator = cadmium::dynamic::translate::make_dynamic_atomic_model
+            <ApplicationGen, TIME,const char* >
+                ("generator" , std::move(i_input_data_control));
+
+    /********************************************/
+    /****** RECIEVER *******************/
+    /********************************************/
+
+    std::shared_ptr<cadmium::dynamic::modeling::model>
+        receiver1 = cadmium::dynamic::translate::make_dynamic_atomic_model
+            <Receiver, TIME>("receiver1");
+
+
+    /************************/
+    /*******TOP MODEL********/
+    /************************/
+    cadmium::dynamic::modeling::Ports iports_TOP = {
 
-/********************************************/
-/****** RECIEVER *******************/
-/********************************************/
+    };
+    cadmium::dynamic::modeling::Ports oports_TOP = {
+            typeid(outp)
```

Comment:
I think there are two tabs before typeid(outp). 
one tab can be ignored before typeid(outp)

Reply:
Solved Thanks for noticing that